gtfs_funnel/schedule_stats_by_route_direction.py

```diff
@@ -12,77 +12,68 @@
-        "stop_sequence",
-        "schedule_gtfs_dataset_key",
-        "trip_instance_key",
-        "shape_array_key",
-        "stop_name",
-        "prior_stop_sequence",
-        "subseq_stop_sequence",
-        "stop_pair",
-        "stop_pair_name",
-        "stop_primary_direction",
-        "stop_meters",
-    ]
+    STOP_TIMES_FILE = dict_inputs.rt_vs_schedule_tables.stop_times_direction
     
-    # Keep only a subset
-    stop_times_gdf2 = stop_times_gdf[stop_times_col]
+    stop_times_gdf = pd.read_parquet(
+    f"{RT_SCHED_GCS}{STOP_TIMES_FILE}_{analysis_date}.parquet",
+    filters=[[("stop_primary_direction", "!=", "Unknown")]
+    ])
+    
+    trip_scheduled_col = [
+    "route_id",
+    "trip_instance_key",
+    "gtfs_dataset_key",
+    "shape_array_key",
+    "direction_id",
+    "route_long_name",
+    "route_short_name",
+    "route_desc",
+    "name"
+    ]
+        
+    trips_df = helpers.import_scheduled_trips(analysis_date, 
+                                             columns = trip_scheduled_col,
+                                             get_pandas = True)
 
+    
     # Merge dfs
     merge_cols = ["trip_instance_key", 
                   "schedule_gtfs_dataset_key", 
                   "shape_array_key"]
-
-    df1 = pd.merge(stop_times_gdf2, trips_to_route_df, on=merge_cols, how="inner")
+    
+    stop_times_with_trip = pd.merge(stop_times_gdf, trips_df, on = merge_cols)
     
     # Fill in missing direction id with 0, per our usual practice.
-    df1.direction_id = df1.direction_id.fillna(0)
+    stop_times_with_trip.direction_id = stop_times_with_trip.direction_id.fillna(0)
+    
+    main_cols = [
+        "route_id",
+        "schedule_gtfs_dataset_key",
+        "direction_id"
+    ]
     
     agg1 = (
-        df1.groupby(
-            [
-                "route_id",
-                "schedule_gtfs_dataset_key",
-                "direction_id",
-                "stop_primary_direction",
-            ]
+        stop_times_with_trip.groupby(
+            main_cols + ["stop_primary_direction"]
         )
         .agg({"stop_sequence": "count"})
         .reset_index()
         .rename(columns={"stop_sequence": "total_stops"})
     )
     
-   # Delete out unknowns
-    agg1 = agg1.loc[agg1.stop_primary_direction != "Unknown"]
-
     # Sort and drop duplicates so that the
     # largest # of stops by stop_primary_direction is at the top
     agg2 = agg1.sort_values(
-        by=["route_id", 
-            "schedule_gtfs_dataset_key",
-            "direction_id",
-            "total_stops"],
+        by= main_cols + ["total_stops"],
         ascending=[True, True, True, False],
     )
-
+    
     # Drop duplicates so only the top stop_primary_direction is kept.
-    agg3 = agg2.drop_duplicates(
-    subset=[
-        "route_id",
-        "schedule_gtfs_dataset_key",
-        "direction_id",
-    ]).reset_index(drop=True)
+    agg3 = agg2.drop_duplicates(subset= main_cols).reset_index(drop=True)
```

Comment:
Combine with above and chain! `agg2 = agg1.sort_values(stuff).drop_duplicates(stuff).reset_index()`